colorcoder/MajorColor.java

```
@@ -0,0 +1,28 @@
+package colorcoder;
+
+enum MajorColor {
+                 WHITE(0),
+                 RED(1),
+                 BLACK(2),
+                 YELLOW(3),
+                 VIOLET(4);
+
+  private final int index;
+
+  private MajorColor(final int index) {
+    this.index = index;
+  }
+
+  int getIndex() {
+    return this.index;
+  }
+
+  public static MajorColor fromIndex(final int index) {
```

This function is similar to the fromIndex present in `MinorColor`. It means any change (handling of invalid index - needing to log or throw exception instead of `return null` - will repeat in two places. Is there a way to prevent this?